translate/Main.hs

```diff
@@ -114,6 +114,36 @@ doTranslate signs circuit = do
             when (undef  /= []) $ putStr $ "The following signals have undefined initial states: \n"
                                     ++ unlines (map show undef) ++ "\n"
 
+handleArcs :: [([Transition DynSignal], Transition DynSignal)] -> [String]
+handleArcs arcLists
+    | arcLists == [] = []
+    | otherwise = do
```

Comment:
Why are you using `do`-notation here? Looks like you can replace this by `otherwise = addSymbTransition ... where ...` dropping a lot of `let`s.